### PR TITLE
Add support for grouping API

### PR DESCRIPTION
**Pull request by v-excelsior**

### CODE REVIEW

File: test/unit/specs/data/group.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
+import { expect } from 'chai'
+import { describe, it } from 'mocha'
+
+import Backendless, { APP_PATH, forTest, prepareMockRequest } from '../../helpers/sandbox'
+
+const createRequestBody = (body) => {
```
Comment (Valodya) [inline test/unit/specs/data/group.js:6]: I believe it can be removed

File: test/unit/specs/data/query-builder.js [hunk 2]
```diff
@@ -264,6 +264,12 @@ describe('<Data> Query Builder', function() {
     expect(() => query.setOffset(-1)).to.throw(errorMsg)
   })
 
+  it('fails when distinct is invalid', function() {
+    const errorMsg = 'Distinct must be a boolean value.'
+
+    expect(() => query.setDistinct('foobar')).to.throw(errorMsg)
```
Comment (Valodya) [inline test/unit/specs/data/query-builder.js:270]: what about other types?

Comment (v-excelsior) [reply test/unit/specs/data/query-builder.js:270]: When I try to invoke API calls with `distinct = 'foobar'` that works the same as `false`. So I think this error makes sense.

Comment (Valodya) [reply test/unit/specs/data/query-builder.js:270]: no, I meant why don't you check this method with other types such as `number` `array` etc?

File: src/urls.js [hunk 3]
```diff
@@ -97,6 +97,10 @@ export default class Urls {
     return `${this.root()}/data`
   }
 
+  group(sourceName) {
```
Comment (Valodya) [inline src/urls.js:100]: `group` -> `dataGrouping`

Comment (Valodya) [reply src/urls.js:100]: `sourceName` -> `tableName`

File: backendless.d.ts [hunk 4]
```diff
@@ -1009,6 +1046,10 @@ declare module Backendless {
 
         find<T = object>(obj?: Backendless.DataQueryBuilder | DataQueryI): Promise<Array<T>>;
 
+        group<T = object>(obj?: Backendless.GroupQueryBuilder | GroupQueryI): Promise<Array<T>>;
+
+        countInGroup<T = object>(obj?: Backendless.GroupQueryBuilder | GroupQueryI): Promise<number>;
```
Comment (Valodya) [inline backendless.d.ts:1051]: I believe in the method `<T = object>` is redundant

File: backendless.d.ts [hunk 5]
```diff
@@ -774,6 +781,36 @@ declare module Backendless {
...
+
+        setRecordsPageSize(recordsPageSize: number): this;
+
+        getRecordsPageSize(): number;
+
+        setGroupDepth(groupDepth: number): this;
+
+        getGroupDepth(): number;
+
+        addGroupPath(groupPath: Object): this;
```
Comment (Valodya) [inline backendless.d.ts:805]: addGroupPath(groupPath: object): this;

File: backendless.d.ts [hunk 6]
```diff
@@ -774,6 +781,36 @@ declare module Backendless {
...
+
+        setGroupPath(groupPath: Object | Array<object>): this;
```
Comment (Valodya) [inline backendless.d.ts:807]: `setGroupPath(groupPath: object | Array<object>): this;`

**Pull request merged**